src/main/java/org/spongepowered/api/command/args/CommandContext.java

```diff
@@ -164,4 +234,16 @@ public boolean hasAny(String key) {
     public boolean hasAny(Text key) {
         return hasAny(ArgUtils.textToArgKey(key));
     }
+
+    /**
+     * A snapshot of the {@link CommandContext}.
+     */
+    static final class Snapshot {
+
+        private final Multimap<String, Object> parsedArgs;
+
+        Snapshot(Multimap<String, Object> parsedArgs) {
+            this.parsedArgs = parsedArgs;
```

Comment:
IMO the defensive copy should be created here instead to avoid potentially bad calls without copy.
Is there an immutable version of this as well?